### DISCUSSION TITLE
A way to auto disable/disconnect wifi when leaving Tesla

### DISCUSSION THREAD

**Discussion by mouragio** [tesla-android/issue-tracker#290] [Q&A]
Hi,

I finally have the PI with Teslaandroid set up properly and works as it should.

The thing I have trouble with is once the PI stays connected to the WIFI of the tesla; external access is impossible. 

Did someone find a solution to have an automation like 
Tesla doors locking -> wifi off (or disconnect wifi)
Tesla doors open -> wifi enabled (connect wifi)

yesterday when scrolling through some tesla apps for android and saw something like an tesla automator which can be used with tasker.
In my head this could work. Obviously wifi on the Tesla cannot be disabled via android; but the wifi on the Pi can?

Maybe someone has another solution. I also use Home assistant to pull data; that one also doesn't work now :)

**Comment by mouragio** [comment]
never mind.. dont think my suggestion will work because also andoid auto will not have a connection to the car to sense the door is open or not.

**Comment by earthwormjimwow** [comment]
You could probably set up an Android automation app to turn off the WiFi hotspot based on whether or not the Pi has an active Bluetooth connection to your phone.  

Maybe set it up so the automation script is only active X number of minutes after a boot up?  That way if the Bluetooth connection fails, or you change phones, you can still access the Pi over WiFi for a few minutes, to make changes.  But in normal use, the script will kick in after you've been driving, then park the car and walk away.

**Comment by mouragio** [comment]
tried that but the wifi hotspot looks like its outside the Android settings; so doesnt get applied when android turns off wifi

thanx for the tip tho

**Reply by earthwormjimwow** [reply]
The hotspot is definitely an Android system setting.  There's nothing outside of Android, since Android is the only OS running.

I used to use this method, but it was unreliable when using 2.4Ghz wifi.  Bluetooth would cut out, turning off the hotspot.   With the 5Ghz wifi option in the latest releases, it should be reliable now.
